app/helpers.php

```diff
@@ -1847,3 +1848,21 @@ function unmix(string $resource): HtmlString
 {
     return app('assets-manifest')->src($resource);
 }
+
+/**
+ * Get an instance of the named migration.
+ */
+function migration(string $name): object
+{
+    $instance = app('files')->getRequire(
+        database_path('migrations').DIRECTORY_SEPARATOR."{$name}.php",
+    );
+
+    if (!$instance instanceof Migration) {
+        throw new Exception(
+            "\"{$name}\" does not return an instance of ".Migration::class.'.',
+        );
+    }
+
+    return $instance;
+}
```

Comment:
that's a lot of fluff just for this

```php
function migration(string $name): Migration
{
    return require database_path("migrations/{$name}.php");
}
```

- getRequire is just a fancy wrapper over plain require. The only benefit from it I can see is it throws full path?
- php already has way to check return type
- as for directory separator thing, just use `/`. It'll be fine

Reply:
return type i couldnt do `Migration` because up() and down() aren't actually methods on that. but okay ill change the rest. i was just trying to follow as close as possible to what the migrator was doing normally

Reply:
but the check here isn't checking for the methods either...? what's the difference between this check with return your check

Reply:
oh my bad it was just a linter getting mad at me for `migration('...')->up()`. i guess i never actually tested it with Migration return type...

will change to what you wrote then :ok_hand: